### PR TITLE
DDLS-107 update the ga workflows to use more shared components

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
Use some of the nicer shared components from our ministryofjustice repo
Fixes DDLS-107

## Approach
There are a few shared components that either replace or enhance some existing workflow functionality. The 3 that I am using here are:

versioning (use semver and have releases)
workspace manager (less clunky than doing it manually)
tf version (hides the clunky logic)

## Learning
NA

## Checklist
- [ ] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes
- [ ] I have checked my work for potential security issues and refered to the [OWASP top 10](https://owasp.org/www-project-top-ten/)

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: .github/workflows/workflow-path-to-live.yml [hunk 1]
```diff
@@ -30,13 +30,15 @@ jobs:
...
+      version_tag: ${{steps.semver_tag.outputs.tag}}
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # pin@v3
-      - name: extract variables for workflow
-        id: variables
-        run: |
-          echo "short_sha=$(echo ${GITHUB_SHA:0:7})" >> $GITHUB_OUTPUT
+      - name: generate semver tag and release
+        id: semver_tag
+        uses: ministryofjustice/opg-github-actions/.github/actions/semver-tag@v2.3.2
```
Comment (charlesmmarshall) [inline .github/workflows/workflow-path-to-live.yml:38]: @jamesrwarren i'm just about to push a newer version that fixes a bug when remotely calling the semver tag action  :)

**Review by xaithe — DISMISSED** [review]

File: .github/workflows/workflow-path-to-live.yml (hunk 1 above)
Comment (jamesrwarren) [reply .github/workflows/workflow-path-to-live.yml:38]: ah ok i'll wait then :)

Comment (charlesmmarshall) [reply .github/workflows/workflow-path-to-live.yml:38]: @jamesrwarren [v2.4.0](https://github.com/ministryofjustice/opg-github-actions/releases/tag/v2.4.0) is now released :)

File: .github/workflows/workflow-path-to-live.yml [hunk 2]
```diff
@@ -30,13 +30,15 @@ jobs:
     name: output workflow variables
     outputs:
       build_identifier: "main"
-      short_sha: ${{ steps.variables.outputs.short_sha }}
+      version_tag: ${{steps.semver_tag.outputs.tag}}
```
Comment (charlesmmarshall) [inline .github/workflows/workflow-path-to-live.yml:33]: oh, the v2+ versions use `created_tag` instead of `tag`

**Review by charlesmmarshall — APPROVED** [review]

**Pull request merged**